rgo/component.py

```diff
@@ -95,94 +93,31 @@ def make_srpm(self, tmpdir):
 
-        spec_version = rpmspec.sourceHeader["Version"]
-        if isinstance(spec_version, bytes):
-            spec_version = spec_version.decode("utf-8")
-
-        _spec_path = os.path.join(workdir, "{!s}.spec".format(spec_name))
+        self.srpm = os.path.join(tmpdir, os.path.basename(srpm_path))
+        shutil.move(srpm_path, self.srpm)
+        LOGGER.info("Built (no)source RPM: %r from %r", self.srpm, self.git.ref_info())
 
-        if self.git:
-            version, release = self.git.describe(self.name, spec_version=spec_version, version_from=self.version_from)
+        # Build overrides srpms
+        for override_git in self.overrides:
+            spec_name = override_git.spec_path or "{!s}.spec".format(self.name)
+            original_spec_path = utils.extract_specfile_from_branch(workdir, override_git, spec_name)
+            final_spec_path = utils.prepare_spec_and_archive(workdir, original_spec_path, self.git, self.name,
+                                                             self.version_from, override_git.patches)
+            if override_git.patches == PatchesAction.keep:
+                utils.prepare_specifle_patches(workdir, final_spec_path, override_git)
+            srpm_path = utils.rpmbuild_srpm(workdir, final_spec_path)
 
-            # Prepare archive
-            if release == "1":
-                # tagged version, no need to add useless numbers
-                nvr = "{!s}-{!s}".format(self.name, version)
-            else:
-                nvr = "{!s}-{!s}-{!s}".format(self.name, version, release)
-            archive = os.path.join(workdir, "{!s}.tar.xz".format(nvr))
-            LOGGER.debug("Preparing archive %r", archive)
-            proc = subprocess.run(["git", "archive", "--prefix={!s}/".format(nvr),
-                                   "--format=tar", self.git.ref],
-                                  cwd=self.git.cwd, check=True,
-                                  stdout=subprocess.PIPE)
-            with open(archive, "w") as f_archive:
-                subprocess.run(["xz", "-z", "--threads=0", "-"],
-                               check=True, input=proc.stdout, stdout=f_archive)
-
-            # Prepare new spec
-            with open(_spec_path, "w") as specfile:
-                prepared = "{!s}\n{!s}".format(
-                    utils.prepare_spec(spec, version, release, nvr, patches),
-                    utils.generate_changelog(self.git.timestamp, version, release, self.git.get_rpm_changelog()))
-                specfile.write(prepared)
-        else:
-            # Just copy spec from distgit
-            shutil.copy2(spec, _spec_path)
-        spec = _spec_path
-
-        _sources = []
-        for src, _, src_type in rpm.spec(spec).sources:
-            if src_type == rpm.RPMBUILD_ISPATCH:
-                if patches == PatchesAction.keep:
-                    _sources.append(src)
-            elif src_type == rpm.RPMBUILD_ISSOURCE:
```

Comment:
> This commit moves around a lot of code, I haven't really gone over all of it to check if it still does the same, but I noticed you've removed this part completely. IIUC this handles secondary sources listed in the spec file which is a marginal case, the main sources are supposed to be in the git and this is rarely used, correct?

ah, I don't know, I didn't know secondary sources were a thing, I though its some not-working version of using tarball from distgit. Although even the secondary sources wouldn't work unless they were present in the distgit itself? or I have missed where they are downloaded.

> It's good to point these things out in the commit message with a brief justification.
> 
> Any other "simplifications" you've done that I've missed? slightly_smiling_face
My bad. I can think of only the way I changed keeping track of the build status, storing the `build_proxy` instead of the `state` and `success`, looking at it now that could be also a separate commit.

Reply:
> ah, I don't know, I didn't know secondary sources were a thing, I though its some not-working version of using tarball from distgit. Although even the secondary sources wouldn't work unless they were present in the distgit itself? or I have missed where they are downloaded.

I'm not sure, I'm fine dropping it, as we'd need to go out of our way to test it and yeah it probably wasn't working in the first place?

Reply:
ok, ok.
I have split the commit into two, it should be more manageable now.